dist/web3-light.js

```diff
@@ -1821,6 +1821,56 @@ var contract = function (abi) {
 };
 
 /**
+ * Should be called to create new ContractFactory
+ *
+ * @method checkForContractAddress
+ * @param {Object} contract
+ * @param {Function} callback
+ * @returns {Undefined}
+ */
+var checkForContractAddress = function(contract, callback){
+    var count = 0;
+
+    // wait for receipt
+    var filter = web3.eth.filter('latest', function(e){
+        if(!e) {
+            count++;
+
+            // stop watching after 50 blocks (timeout)
+            if(count > 50) {
+                if(callback)
+                    callback(new Error('Contract couldn\'t be deployed'));
+
+                filter.stopWatching();
+
+            } else {
+
+                web3.eth.getTransactionReceipt(contract.transactionHash, function(e, receipt){
+                    if(receipt) {
+
+                        web3.eth.getCode(receipt.contractAddress, function(e, code){
```

Comment:
Yes this is necessary, as the contract address can be created and the code executed, but there was not enough gas to store the code.

I had these issues see here: https://github.com/ethereum/yellowpaper/issues/132